docs(idl): correct PC references to $pc builtin in examples (#2030)
riscv-software-src/riscv-unified-db: doc/docs/idl/for-spec-writers.mdx

## Writing CSR callbacks

CSR YAML definitions can include `sw_read()` and `sw_write()` callbacks for custom read and write behavior. Note that while `sw_read()` can be defined at the CSR or field level, `sw_write()` is defined **per-field** (operating on an individual CSR field).

### `sw_read()`

> Called when software reads the CSR. Returns the value software should see:

Reviewer: Please restore the sentence about "stored state". Thanks.

```yaml
# sw_read() for a CSR:
sw_read(): |
  return $bits(CSR[mip]) & $bits(CSR[mideleg]);
```

> ### `sw_write()`

Reviewer: ```suggestion
### `sw_write(csr_value)`
```

> Defined per-field; called when software writes to a field. The `val` parameter receives the value written by software. Apply WARL (Write Any Read Legal) filtering here:

Reviewer: ```suggestion
Defined per field. Called when software writes the CSR. The `csr_value` parameter receives the value written by software and individual fields can be accessed like a CSR, e.g. `csr_value.FIELD`. Apply WARL (Write Any Read Legal) filtering here:
```

```yaml
# In a field definition, sw_write() for a WARL field:
sw_write(val): |
  if (val <= 1'b1) {

>     return val;

Reviewer: ```suggestion
sw_write(csr_value): |
  # accept values [ 0, 1 ] only, otherwise ignore
  if (csr_value.FIELD <= 1'b1) {
    return csr_value.FIELD;
```
